src/universe.rs

```diff
@@ -238,3 +324,48 @@ impl Default for RuleSet {
         Self::new()
     }
 }
+
+#[cfg(test)]
+mod test {
+    use crate::{ast::Sym, Symbols};
+
+    use super::{SymbolOverlay, SymbolStorage, SymbolStore};
+
+    #[test]
+    fn overlay() {
+        let mut plain = SymbolStore::new();
+        plain.insert_unnamed();
+        plain.insert_unnamed();
+        plain.insert_unnamed();
+        plain.get_or_insert_named("a");
+        let overlay_copy = plain.clone();
+        let mut overlaid = SymbolOverlay::new(&overlay_copy);
+
+        assert_eq!(
+            plain.get_or_insert_named("b"),
+            overlaid.get_or_insert_named("b")
+        );
+
+        assert_eq!(
+            plain.get_or_insert_named("c"),
+            overlaid.get_or_insert_named("c")
+        );
```

Comment:
These are true, but what are you intending to test with these? They don't seem meaningful properties to me:

What we do want is that the already existing symbol `"a"`  is resolved to the same `Sym` in both `plain` and `overlay`, but that some new symbol `"b"` gets the same `Sym` in both is merely coincidental.

What should be tested is that new symbols added in the overlay don't collide with symbols from the underlying storage, and that the names of those new symbols can be obtained correctly (i.e. that the subtraction/offsetting logic is working correctly).

Reply:
You're right, but I did it for two reasons:
- it's easier
- it makes debugging easier when the behaviour matches

Being overly strict only becomes a problem when there's a need to build some unusual overlay, and at the moment I can't even think about what that could be, so I decided to make it work and worry later.

Reply:
Perhaps for now you can just add a comment there that we test that the overlay store behaves just as if the symbols had been added to the underlying one directly for now?

Then we can leave any generalizations for later, indeed.

Reply:
Done.